targets/TARGET_STM/stm_spi_api.c

```diff
@@ -602,6 +606,25 @@ static const uint32_t baudrate_prescaler_table[] =  {SPI_BAUDRATEPRESCALER_2,
                                                      SPI_BAUDRATEPRESCALER_256
                                                     };
 
+static uint8_t spi_get_baudrate_prescaler_rank(uint32_t value)
+{
+#if TARGET_STM32H7
+    return value >> 28 & 0x07;
+#else /* TARGET_STM32H7 */
+    return value >> 3 & 0x07;
+#endif /* TARGET_STM32H7 */
+}
+
+static int spi_get_baudrate(spi_t *obj)
+{
+    struct spi_s *spiobj = SPI_S(obj);
+    SPI_HandleTypeDef *handle = &(spiobj->handle);
+
+    int freq = spi_get_clock_freq(obj);
+    uint8_t baudrate_rank = spi_get_baudrate_prescaler_rank(handle->Init.BaudRatePrescaler);
```

Comment:
Isn't that possible to re-use LL macros like LL_SPI_GetBaudRatePrescaler or HAL equivalent ?

Reply:
`LL_SPI_GetBaudRatePrescaler` returns some prescaler constant, but not prescaler value itself. So extra code that converts this constant to prescaler rank itself is needed.

Reply:
ok !